### PR TITLE
Document new configuration to enable ssl for ingesting via mqtt.

### PR DESCRIPTION
**Pull request by andreidan**

### CODE REVIEW

File: blackbox/docs/administration/ssl.txt [hunk 1]
```diff
@@ -9,14 +9,18 @@ Secured Communications (SSL/TLS)
    *Secured Communications* is an Enterprise Edition feature.
 
 Secured communication allows you to encrypt traffic between the CrateDB node
-and a client. This applies to connections using HTTP (i.e. Admin UI, Crash, Rest)
-as well as connections using the PostgreSQL Wire Protocol (i.e. JDBC, psql).
+and a client. This applies to connections using HTTP (i.e. Admin UI, Crash, Rest),
+connections using the PostgreSQL Wire Protocol (i.e. JDBC, psql), and MQTT 
+connections (i.e. MQTT ingestion source)
```
Comment (m-kharbat) [inline blackbox/docs/administration/ssl.txt]: link to the ingestion documentation ?

Comment (mxm) [inline blackbox/docs/administration/ssl.txt]: How about:

> This applies to connections using HTTP (i.e. Admin UI, Crash, Rest), the PostgreSQL Wire Protocol (i.e. JDBC, psql), and MQTT (i.e. MQTT ingestion source).

File: blackbox/docs/administration/ssl.txt [hunk 2]
```diff
@@ -9,14 +9,18 @@ Secured Communications (SSL/TLS)
...
+
 Connections are secured using Transport Layer Security (TLS).
 
 Note that once SSL is enabled for HTTP connections, only connections using
-HTTPS are allowed. This is in contrast to the PostgreSQL Wire Protocol which
-still allows non-encrypted connections when SSL is enabled. If you want to
-enforce SSL usage, please have a look at :ref:`administration_hba`.
+HTTPS are allowed. MQTT secure connections are also exclusive and once enabled, 
+unsecure connections will not be possible.
```
Comment (mxm) [inline blackbox/docs/administration/ssl.txt]: Would remove the "exclusive", might be confusing to users:

> "Same applies to secure MQTT connections."

File: blackbox/docs/administration/ssl.txt [hunk 3]
```diff
@@ -179,6 +184,17 @@ information including the different SSL modes please visit the
...
+------------------------------------------------
+A CrateDB node configured to run as :ref:`MQTT broker <ingest_mqtt>` can 
+accept encrypted connections if ssl ingestion is enabled, namely if the 
+``ssl.ingestion.mqtt.enabled`` configuration is set to ``true`` and the 
+configured port ``ingestion.mqtt.port`` is ``8883``.
+
+There are various MQTT client implementations available and  whichever you will 
+use, you will need to configure the Certificate Authorities that are to be 
+trusted by the client and the certificate that will be used for TLS based 
+authentication.
```
Comment (mxm) [inline blackbox/docs/administration/ssl.txt]: Not necessarily, the trust store is optional. Should we skip this paragraph and just link to the SSL configuration page?

Comment (andreidan) [reply blackbox/docs/administration/ssl.txt]: Hmm, this is the SSL configuration page, but you are right that this makes it sound mandatory.

File: blackbox/docs/administration/ssl.txt [hunk 4]
```diff
@@ -11,13 +11,12 @@ Secured Communications (SSL/TLS)
 Secured communication allows you to encrypt traffic between the CrateDB node
 and a client. This applies to connections using HTTP (i.e. Admin UI, Crash, Rest),
 connections using the PostgreSQL Wire Protocol (i.e. JDBC, psql), and MQTT 
-connections (i.e. MQTT ingestion source)
+(i.e. :ref:`MQTT ingestion source <ingest_mqtt>`)
```
Comment (mxm) [inline blackbox/docs/administration/ssl.txt]: Minor refinement to remove another "connections":

> This applies to connections using HTTP (i.e. Admin UI, Crash, Rest),
~~connections using~~ the PostgreSQL Wire Protocol (i.e. JDBC, psql), and MQTT
(i.e. :ref:`MQTT ingestion source <ingest_mqtt>`)**.**

**Pull request merged**